pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -426,7 +425,7 @@ public PersistentTopic(String topic, ManagedLedger ledger, BrokerService brokerS
             this.transactionBuffer = new TransactionBufferDisable(this);
         }
         transactionBuffer.syncMaxReadPositionForNormalPublish(ledger.getLastConfirmedEntry(), true);
-        if (ledger instanceof ShadowManagedLedgerImpl) {
+        if (ledger.getConfig().getShadowSource() != null) {
```

Comment:
Use `if (!StringUtils.isBlanck(ledger.getConfig().getShadowSource()))`?

Reply:
No need here because exception will be thrown in `TopicName.get` if the shadow source is invalid (including blank string). Normally, the shadow source could only be set by `createShadowTopic` so the shadow source should always be a valid topic name or null unless the metadata store is hacked.

Reply:
Ok